src/uu/nice/src/nice.rs

```diff
@@ -30,8 +30,75 @@ const ABOUT: &str = "\
     process).";
 const USAGE: &str = "{} [OPTIONS] [COMMAND [ARGS]]";
 
+/// Transform legacy arguments into a standardized form.
+///
+/// The following are all legal argument sequences to GNU nice:
+/// - "-1"
+/// - "-n1"
+/// - "-+1"
+/// - "--1"
+/// - "-n -1"
+///
+/// It looks initially like we could add handling for "-{i}", "--{i}"
+/// and "-+{i}" for integers {i} and process them normally using clap.
+/// However, the meaning of "-1", for example, changes depending on
+/// its context with legacy argument parsing. clap will not prioritize
+/// hyphenated values to previous arguments over matching a known
+/// argument.  So "-n" "-1" in this case is picked up as two
+/// arguments, not one argument with a value.
+///
+/// Given this context dependency, and the deep hole we end up digging
+/// with clap in this case, it's much simpler to just normalize the
+/// arguments to nice before clap starts work. Here, we insert a
+/// prefix of "-n" onto all arguments of the form "-{i}", "--{i}" and
+/// "-+{i}" which are not already preceded by "-n".
+fn standardize_nice_args(args: impl uucore::Args) -> impl uucore::Args {
+    let mut v = Vec::<OsString>::new();
+    let mut saw_n = false;
+    let mut saw_command = false;
+    for s in args {
+        if saw_command {
+            v.push(s);
+        } else if s == <str as AsRef<OsStr>>::as_ref("-n")
+            || s == <str as AsRef<OsStr>>::as_ref("--adjustment")
```

Comment:
`OsStr::new("-n")` might be a bit easier to read.

Also any unambiguous prefix of `--adjustment` (e.g. `--a`, `--adj`, etc.) should all be accepted as well.